src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -218,6 +215,21 @@ def start():
     _session.start_time = datetime.datetime.now()
 
 
+@decorators.suppress_all_exceptions(raise_in_diagnostics=True)
+def flush():
+    global _session  # pylint: disable=global-statement
```

Comment:
Can't you reset the current session instead of creating a new _session? `global` must not be used.